contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -504,7 +553,7 @@ contract VaultAdmin is VaultStorage {
         if (tokenAsset.decimals != 0) {
             return;
         }
-        uint256 decimals = IBasicToken(token).decimals();
+        uint8 decimals = IBasicToken(token).decimals();
```

Comment:
Does cause a gas regression? Sometimes it's cheaper to work with uint256s since they match up to a full slot.

That said, this method is only rarely going to be called, so it's probably not worth gas optimizing.

Reply:
Yes, it is cheaper to do arithmetic and pass numbers around between functions using uint256.
But in this case we are getting ready to write the decimal value to storage. We get huge gas savings in other functions by packing the asset data into a single slot.